ansible_runner/config/runner.py

```diff
@@ -243,6 +243,13 @@ def prepare_env(self):
         self.env["RUNNER_OMIT_EVENTS"] = str(self.omit_event_data)
         self.env["RUNNER_ONLY_FAILED_EVENTS"] = str(self.only_failed_event_data)
 
+        original_stdout_callback = os.getenv('ANSIBLE_STDOUT_CALLBACK')
```

Comment:
I'm still trying to grasp everything this PR is doing, but shouldn't we check `self.env` here instead of using `os.getenv()`?

Looking at `BaseConfig._prepare_env()`, it populates `self.env` with the environment variables, but also populates with any custom env vars (e.g. from `env/envvars`). So we'd be missing the latter with the current line of code.

Reply:
I agree with you. In another comment, I argued that `ANSIBLE_STDOUT_CALLBACK` should still be set in the RunnerConfig class instead of the runner.py module. The typical practice there is to compare with `self.env` as opposed to `os.environ`.

> Looking at `BaseConfig._prepare_env()`, it populates `self.env` with the environment variables

That's only true if the run is not containerized. For reasons of repeatability and functionality, the environment for runs in container start from scratch. If the run is _not_ containerized, then `self.env` should already be updated with `os.environ`. Thus, we only need consider `self.env`.